From 6e140a4eed5e135bb49710629d579c871adfa0bd Mon Sep 17 00:00:00 2001
From: Yusuke Goda <yusuke.goda.sx@renesas.com>
Date: Thu, 17 Jan 2019 06:04:21 +0000
Subject: [PATCH 769/909] arm64: dts: r8a77965-m3nulcb: Fix source clock for DU

commit 44145606a843a196841d9b4d4b3669f7feae38a8 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

ULCB Versaclock default output clock is based on its ROM code,
and ULCB case is 33HMz.
But current DT is missing such information, and Versaclock driver
can't handle it today.

We need to update Versaclock driver to handle it correctly.
But as Quick-Hack, this patch indicates it via versaclock5_out3
temporally.

Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a77965-m3nulcb.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77965-m3nulcb.dts b/arch/arm64/boot/dts/renesas/r8a77965-m3nulcb.dts
index 5158094feaad..0d3a6e65a77e 100644
--- a/arch/arm64/boot/dts/renesas/r8a77965-m3nulcb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77965-m3nulcb.dts
@@ -59,6 +59,13 @@
 	vspm_if {
 		compatible = "renesas,vspm_if";
 	};
+
+	versaclock5_out3: versaclk-3 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* Initial value of versaclock out3 */
+		clock-frequency = <33000000>;
+	};
 };
 
 &du {
@@ -66,7 +73,7 @@
 		 <&cpg CPG_MOD 723>,
 		 <&cpg CPG_MOD 721>,
 		 <&versaclock5 1>,
-		 <&versaclock5 3>,
+		 <&versaclock5_out3>,
 		 <&versaclock5 2>;
 	clock-names = "du.0", "du.1", "du.3",
 		      "dclkin.0", "dclkin.1", "dclkin.3";
-- 
2.17.1

